src/sgx-verifier/AttestationVerifier.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.24;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import {IAttestationVerifier} from "./IAttestationVerifier.sol";
+import {IDcapAttestation} from "./IDcapAttestation.sol";
+
+import {BytesUtils} from "./utils/BytesUtils.sol";
+
+/// @dev This contract is modified from https://github.com/automata-network/scroll-prover/blob/demo/contracts/src/core/AttestationVerifier.sol
+contract AttestationVerifier is Ownable, IAttestationVerifier {
+    using BytesUtils for bytes;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    /**********
+     * Errors *
+     **********/
+
+    /// @dev Thrown when the given attestation report is invalid.
+    error ErrorInvalidReport();
+
+    /// @dev Thrown when the user data from the attestation report mismatch.
+    error ErrorReportDataMismatch();
+
+    /// @dev Thrown when the MrSigner from the attestation report is invalid.
+    error ErrorInvalidMrSigner();
+
+    /// @dev Thrown when the MrSigner from the attestation report is invalid.
+    error ErrorInvalidMrEnclave();
+
+    /***********************
+     * Immutable Variables *
+     ***********************/
+
+    /// @notice The address of automata's DCAP Attestation contract.
+    IDcapAttestation public immutable attestationVerifier;
+
+    /*********************
+     * Storage Variables *
+     *********************/
+
+    /// @dev The list of trusted enclaves.
+    EnumerableSet.Bytes32Set private trustedUserMrEnclave;
```

Comment:
Is this basically the "program hash"? Or the identity of the (Intel) signer?

Reply:
`MrEnclave` is the program hash.